### ISSUE TITLE
Thoughts on how to run iJavaScript on mybinder?

### ISSUE THREAD

**Issue by paulroth3d** [n-riesco/ijavascript#273]
@n-riesco - might you know what would be needed for a notebook to run in mybinder.org - so it would have the ijavaScript kernel running?

Sorry if there are a number of [threads on this topic, and I'm adding to them](https://xkcd.com/927/)

but I couldn't tell if this still needed something and was open [like #196](https://github.com/n-riesco/ijavascript/issues/196),

or well understood by others, and not me yet [like #277](https://github.com/n-riesco/ijavascript/issues/227) or [#188](https://github.com/n-riesco/ijavascript/issues/188)

**Comment by paulroth3d** [comment]
I've been looking briefly into the documentation, from the great [zero-to-binder tutorial](https://github.com/Build-a-binder/build-a-binder.github.io/blob/master/workshop/10-zero-to-binder.md)

and found in their [readme](https://github.com/Build-a-binder/build-a-binder.github.io/blob/master/workshop/10-zero-to-binder.md) that they support things outside of just `requirements.txt` - pointing to docs on [repo2docker](https://repo2docker.readthedocs.io/en/latest/config_files.html)

There it seems like there might be two options:

* Provide a docker file
* Or do something like an `apt.txt` to install node and then run a `postBuild`

I noticed that there was a [pull request #192](https://github.com/n-riesco/ijavascript/pull/192) that seemed accepted ([test/ijavascript-base-notebook-latest/Dockerfile](https://github.com/n-riesco/ijavascript/pull/192)) but then removed.

(...although [it seems there is a ... copy stored in a gist](https://github.com/parmentelat/scratch-js/blob/787d35abf419995952fa2786d7c204998da09cea/nbhosting/Dockerfile))

I was curious what the current thoughts were,
or if there was an image on dockerhub I missed

**Comment by paulroth3d** [comment]
For anyone else looking to run your project on mybinder, I made an instance that fixes a few of the issues I noticed with https://hub.docker.com/r/datainpoint/ijavascript-notebook

Plain:
[![Binder:what can I do with this](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/paulroth3d/jupyter-ijavascript-utils/main)

Pre-loading the example.ipynb
[![Binder:what can I do with this](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/paulroth3d/jupyter-ijavascript-utils/main?labpath=example.ipynb)

Writeup is here: https://github.com/paulroth3d/docker_jupyter-ijavascript-utils

TLDR:

Either make a `DockerFile` similar to the following:

```
# syntax=docker/dockerfile:1

FROM darkbluestudios/jupyter-ijavascript-utils:amd_1.0.1
```

Or, if you'd like to update to have your own npm modules pre-installed:

* make your own Docker image, adding in `package.json` and `package-lock.json`
* Create a `Dockerfile` and [look at this `Dockerfile` for what was used there](https://github.com/paulroth3d/docker_jupyter-ijavascript-utils/blob/main/Dockerfile)

**Comment by paulroth3d** [comment]
@n-riesco - would rather ask - would this be okay to close?

**Comment by n-riesco** [comment]
@paulroth3d First of all, apologies for my silence (I keep forgetting about open issues like yours, when I can't act on them right away :disappointed: ).

I've labelled this issue as a `question` (I'm using this label as temporary FAQ before I rehaul the README).

Thank you for all the work you've put. It looks impressive in many ways. I want to give it more visibility in the README (I will leave this issue open as a reminder for me to do so).
